control-plane-operator/controllers/hostedcontrolplane/configoperator/reconcile.go

```diff
@@ -229,7 +230,8 @@ var (
 	}
 )
 
-func ReconcileDeployment(deployment *appsv1.Deployment, image, hcpName, openShiftVersion, kubeVersion string, ownerRef config.OwnerRef, config *config.DeploymentConfig, availabilityProberImage string, enableCIDebugOutput bool, platformType hyperv1.PlatformType, apiInternalPort *int32, konnectivityAddress string, konnectivityPort int32, oauthAddress string, oauthPort int32, releaseImage string, additionalTrustBundle *corev1.LocalObjectReference) error {
+func ReconcileDeployment(deployment *appsv1.Deployment, image, hcpName, openShiftVersion, kubeVersion string, ownerRef config.OwnerRef, config *config.DeploymentConfig, availabilityProberImage string, enableCIDebugOutput bool, platformType hyperv1.PlatformType, apiInternalPort *int32, konnectivityAddress string, konnectivityPort int32, oauthAddress string, oauthPort int32, releaseImage string, additionalTrustBundle *corev1.LocalObjectReference, hcp *hyperv1.HostedControlPlane) error {
```

Comment:
I'm going to recommend a refactor here in this ReconcileDeployment function that keeps our kubevirt platform codepaths completely independent of the standard HCCO deployment.

Take a look at how the ```if additionalTrustBundle != nil``` functionality. That injects volumes and volumemounts into the deployment when that setting is present. I'd like to see something similar to that for the kubevirt platform that injects our logic into the HCCO deployment after the standard HCCO deployment is already rendered.


so, something like this, right around where the additionalTrustBundle logic is
```
	if additionalTrustBundle != nil {
		util.DeploymentAddTrustBundleVolume(additionalTrustBundle, deployment)
	}

        if isExternalInfraKv(hcp) {
                // injects the kubevirt credentials secret volume, volume mount path, and appends cli arg. 
                util.DeploymentAddKubevirtInfraCredentials(deployment)
        }

```